### PR TITLE
ci(phpstan): check `@throws`

### PR DESCRIPTION
**Pull request by simPod**
Closes #1331

### CODE REVIEW

**Comment by spawnia** [comment]
Please mention me when you are done.

**Comment by simPod** [comment]
Yup

**Comment by simPod** [comment]
@spawnia le mention

**Review by spawnia — COMMENTED** [review]
Thanks for the valiant effort. I am not sure I like the addition of 2000 lines of code and presumably a much higher maintenance effort. A part of me just wants to delete all `@throws` annotations and call it a day.

File: src/Utils/SchemaExtender.php [hunk 1]
```diff
@@ -574,6 +631,10 @@ protected function extendMaybeNamedType(?Type $type = null): ?Type
     /**
      * @param array<DirectiveDefinitionNode> $directiveDefinitions
      *
+     * @throws InvariantViolation
+     * @throws \ReflectionException
+     * @throws \Exception
```
[the remark this answers is not in the document]
Comment (simPod) [reply src/Utils/SchemaExtender.php:636]: Yup, but I did not want to start refactoring the exception flow as well in this PR. But exactly, the value is that now we see that sometimes we throw something we shouldn't.

File: src/Error/CoercionError.php [hunk 2]
```diff
@@ -54,6 +54,9 @@ public function printInputPath(): ?string
         return $path;
     }
 
+    /**
+     * @throws \JsonException
+     */
```
[the remark this answers is not in the document]
Comment (simPod) [reply src/Error/CoercionError.php:59]: I did not want to change the code in this PR though. This is the current state as is. It's like adding types, we add them first and then refactor.

I agree we should then inspect why some exceptions are thrown.

File: src/Server/StandardServer.php [hunk 3]
```diff
@@ -113,6 +123,13 @@ public function executeRequest($parsedBody = null)
      * See `executePsrRequest()` if you prefer to create response yourself
      * (e.g. using specific JsonResponse instance of some framework).
      *
+     * @throws \Exception
+     * @throws \InvalidArgumentException
+     * @throws \JsonException
+     * @throws \RuntimeException
+     * @throws InvariantViolation
+     * @throws RequestError
```
[the remark this answers is not in the document]
Comment (simPod) [reply src/Server/StandardServer.php:131]: I don't use any of those tools tbh. You know cs-fixer better, I don't see any such rule in the docs.
That reminds me I'd like to request such rule for phpcs though.

[the remark this answers is not in the document]
Comment (spawnia) [reply src/Server/StandardServer.php:131]: https://github.com/webonyx/graphql-php/pull/1341

[the remark this answers is not in the document]
Comment (simPod) [reply src/Server/StandardServer.php:131]: I've run the cs fix

File: benchmarks/BuildSchemaBench.php [hunk 4]
```diff
@@ -67,6 +69,13 @@ public function makeSchemaString(): void
         }
     }
 
+    /**
+     * @throws \GraphQL\Error\SyntaxError
+     * @throws \Exception
+     * @throws Error
+     * @throws \ReflectionException
+     * @throws InvariantViolation
+     */
```
[the remark this answers is not in the document]
Comment (simPod) [reply benchmarks/BuildSchemaBench.php]: Reverted and excluded benchmarks

**Comment by spawnia** [comment]
I would like to see the impact a change like https://github.com/webonyx/graphql-php/pull/1343 has on the code base with this static analysis active. Once we merge it into `master`, can you please update this branch with it so I can judge the impact. Please do not use force push for that, I would like to see the diff.

Side note, I am generally not too fond of force pushes in merge requests. It makes reviews harder for me.

**Review by spawnia — APPROVED** [review]
Thanks for your work on this. I will squash and merge when conflicts are resolved.

**Pull request merged**